### PR TITLE
[2641] optimize/reorganize unit tests

### LINKED ISSUE #2641: Refactor and optimize test_parse.py for better maintainability and performance
## Background ##
[`test_parse.py`](https://github.com/raft-tech/TANF-app/blob/develop/tdrs-backend/tdpservice/parsers/test/test_parse.py) has become a large, entropy-heavy file containing overly broad integration-style tests. To improve maintainability, readability, and coverage precision, we need to refactor this test file toward more targeted, unit-like tests. 

The goal is to reduce complexity, minimize redundant or bloated test cases, and increase overall test coverage while aligning with updated testing philosophy and standards.

### Why It Matters ###
- Improves clarity and maintainability of the test suite
- Makes debugging test failures faster and easier
- Increases confidence in test coverage, especially around edge cases and error handling
- Supports better use of CI/CD tools like Codecov and flake8 for quality reporting

### What's Changing ###
- Refactoring large integration-style tests into smaller, focused unit tests
- Standardizing test structure and naming conventions
- Reducing unnecessary test data files and complexity
- Using parameterized test templates for clean, consistent test cases

### Acceptance Criteria ###
- [ ] Reduce complexity, entropy, and line count (measurable via flake8/codecov)
- [ ] Reduce test data file count where possible
- [ ] Increase meaningful test coverage (measurable via Codecov)
- [ ] Refactored tests no longer trigger high error counts or arbitrary complexity flags
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary
- [ ] [Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) has been updated with testing steps & relevant media  

### Usability Testing Criteria ###
_N/A — This is a backend refactor and should not impact external users._

### Notes ###
- There are ~100 unit tests currently — refactor should aim to simplify and clarify without reducing coverage.
- Testing philosophy guidelines:
  * **Base case** — Does it fall over?
  * **Positive case** — Covers `if` branches
  * **Negative case** — Covers `else` branches
  * **Fuzzer/error handling** — Tests exceptions and failure paths
- Example template:
```python
@pytest.mark.parametrize("input, expected", [
    ("abc", 0),
    ("def", 3),
    ("xyz", -1)
])
def test_f1(fixture, input, expected):
    assert f1(input) == expected

def test_f1_negative(fixture, input):
    assert f1(input) == err_obj

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2641 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up --build
cd tdrs-backend && docker-compose up --build
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [ ] Reduce complexity, entropy, and line count (measurable via flake8/codecov)
- [ ] Reduce test data file count where possible
- [ ] Increase meaningful test coverage (measurable via Codecov)
- [ ] Refactored tests no longer trigger high error counts or arbitrary complexity flags
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

[Release digest](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/JSIbkxujKUxhKv2cXFEg/) has been updated with testing steps & relevant media
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/data_files/test/test_admin_filters.py [hunk 1]
```diff
@@ -0,0 +1,194 @@
+"""Tests for admin list filters on DataFiles."""
```
Comment (raftmsohani) [inline tdrs-backend/tdpservice/data_files/test/test_admin_filters.py:1]: NOTE: there are a lot of movements, mostly moving function based test to class based to make it more organized. But I also decided to improve coverage so added some tests for reparse command, etc

**Review by elipe17 — APPROVED** [review]

**Review by mattcoleanderson — APPROVED** [review]
